raiden/waiting.py

```diff
@@ -42,6 +52,184 @@
 TRANSPORT_ERROR_MSG = "Waiting for protocol messages requires a running transport."
 
 
+def _get_channel_state_by_partner_address(
+    chain_state: ChainState,
+    token_network_registry_address: TokenNetworkRegistryAddress,
+    token_address: TokenAddress,
+    partner_address: Address,
+) -> Optional[NettingChannelState]:
+    token_network = views.get_token_network_by_token_address(
+        chain_state=chain_state,
+        token_network_registry_address=token_network_registry_address,
+        token_address=token_address,
+    )
+
+    if token_network is None:
+        raise ValueError(
+            f"The token {to_checksum_address(token_address)} is not registered on "
+            f"the network {to_checksum_address(token_network_registry_address)}."
+        )
+    return views.get_channelstate_by_token_network_and_partner(
+        chain_state, token_network.address, partner_address
+    )
+
+
+class ChannelStateCondition(ABC):
+    @abstractmethod
+    def evaluate_condition(
```

Comment:
Since this method is already on a condition, maybe just use `evaluate`, to make it shorter?